autoload/lsp/ui/vim/folding.vim

```diff
@@ -0,0 +1,183 @@
+let s:folding_ranges = {}
+let s:textprop_name = 'vim-lsp-folding-linenr'
+
+function! s:find_servers() abort
+    return filter(lsp#get_whitelisted_servers(), 'lsp#capabilities#has_folding_range_provider(v:val)')
+endfunction
+
+function! lsp#ui#vim#folding#fold(sync) abort
+    let l:servers = s:find_servers()
+
+    if len(l:servers) == 0
+        call lsp#utils#error('Folding not supported for ' . &filetype)
+        return
+    endif
+
+    let l:server = l:servers[0]
+    call lsp#ui#vim#folding#send_request(l:server, bufnr('%'), a:sync)
+endfunction
+
+function! s:set_textprops(buf) abort
```

Comment:
@thomasfaingnaert  I just got an error about prop being set when the buf gets unloaded. Can we have a check so that if the buf is unloaded we don't do anything?

It happened quite fast when I closed the buffer. A bit difficult to repro but it happens.

Reply:
@prabirshrestha Would a `silent!` before the `call` fix this?

Reply:
That would fix too. but since the buf is already unloaded I think the check would be better. That way it becomes a noop.

Reply:
Adding `if !bufexists(a:buf) | return | endif` might be better